Projects/Server/Serialization/UnmanagedDataReader.cs

```diff
@@ -205,42 +285,54 @@ public Type ReadTypeByHash()
         return t;
     }
 
+    /// <summary>
+    /// Reads the next set of bytes to fill the buffer.
+    /// </summary>
+    /// <param name="buffer">A reference span that will be filled with the next set of bytes.</param>
+    /// <returns>The length of the buffer.</returns>
+    /// <exception cref="OutOfMemoryException">Thrown if the buffer is larger than the remaining data to read in the file.</exception>
     public int Read(Span<byte> buffer)
     {
         var length = buffer.Length;
-        if (length > _size - _position)
+        if (length > _size - Position)
         {
             throw new OutOfMemoryException();
         }
 
-        new ReadOnlySpan<byte>(_ptr + _position, length).CopyTo(buffer);
-        _position += length;
+        new ReadOnlySpan<byte>(_ptr + Position, length).CopyTo(buffer);
+        Position += length;
         return length;
     }
 
+    /// <summary>
+    /// Sets the current position of the stream to a specified value.
+    /// </summary>
+    /// <param name="offset">The new position, relative to the <paramref name="origin"/> parameter.</param>
+    /// <param name="origin">The reference point for the <paramref name="offset"/> parameter. It can be one of the values of <see cref="SeekOrigin"/>.</param>
+    /// <returns>The new position in the stream, in bytes.</returns>
+    /// <exception cref="ArgumentOutOfRangeException">Thrown when the <paramref name="offset"/> or the resulting position is out of the valid range.</exception>
+    /// <exception cref="InvalidOperationException">Thrown if the stream does not support seeking.</exception>
     public virtual long Seek(long offset, SeekOrigin origin)
     {
         Debug.Assert(
-            origin != SeekOrigin.End || offset <= 0 && offset > _size,
+            origin != SeekOrigin.End || (offset <= 0 && offset > _size),
             "Attempting to seek to an invalid position using SeekOrigin.End"
         );
         Debug.Assert(
-            origin != SeekOrigin.Begin || offset >= 0 && offset < _size,
+            origin != SeekOrigin.Begin || (offset >= 0 && offset < _size),
             "Attempting to seek to an invalid position using SeekOrigin.Begin"
         );
         Debug.Assert(
-            origin != SeekOrigin.Current || _position + offset >= 0 && _position + offset < _size,
+            origin != SeekOrigin.Current || (Position + offset >= 0 && Position + offset < _size),
```

Comment:
IDE will complain about unnecessary parenthesis. Same for the other lines where parentheses are added.

Reply:
[RCS1123](https://josefpihrt.github.io/docs/roslynator/analyzers/RCS1123/)

You have it marked in the linting. 

I recommend changing the style if you're doing something different locally than the repo. IDE complains to add them.

Reply:
They aren't necessary though? I am confused sorry.